### DISCUSSION TITLE
Platform Automerge not reenabled after conflicts

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#21709] [Suggest an Idea]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.108.0

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.9.x

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

When (platform) auto merge is enabled for dependencies on adjacent lines, if both lines are updated, it will take three runs of Renovate to merge both the changes in:

1. During the first run, a PR will be created for both, and both will be set to automerge. If we wait a little while, one of them will succeed first, be automerged by the platform into the base branch, and thereby conflict the other PR.
2. On the next run of Renovate, it will come across the previous PR, identify that it is conflicted, and rebase the PR. In the process, Renovate or the platform will abort the auto merge because of the source branch changes. The CI pipeline will run again, but once it's done, the platform will not automerge.
3. It takes a third time to scan the project, and notice that the previous pipeline was successful can be merged by Renovate.

![image](https://user-images.githubusercontent.com/1637358/180570592-40b439ad-36b8-4bc2-8eb1-2055953a474a.png)

![image](https://user-images.githubusercontent.com/1637358/180570522-77f6e291-4c27-4dad-abcc-dc73037adf06.png)

I would expect that, on rebase, the platform automerge flag is reenabled, so that there's no need for three runs to merge two dependency updates.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
i think this is not a bug, as platform automerge is only used at PR creation for now

**Comment by HonkingGoose** [comment]
Our [`platformAutomerge` docs](https://docs.renovatebot.com/configuration-options/#platformautomerge) say:

> Renovate tries platform-native automerge only when it initially creates the PR. Any PR that is being updated will be automerged with the Renovate-based automerge.

**Comment by astellingwerf** [comment]
Would you care to explain that decision? Why do we require two extra runs for every conflict?

**Comment by rarkins** [comment]
We are not "requiring" anything at all. Right now, platform automerge can be disabled by the platform when we force push. "Not taking the necessary steps to re-enable platform automerge after a branch update" is an entirely different thing to "intentionally requiring an extra run just because we felt like it". 

You can go back to where this feature was added for gitlab and see if it was thought of at the time. Maybe it was an oversight, or maybe it's difficult or inefficient. I don't recall. I don't think anyone objects to it in principle but have no idea if it can be done easily or efficiently.
